tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -380,23 +380,26 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
                 var samplingPriority = DistributedTracer.Instance.GetSamplingPriority();
                 traceContext.SetSamplingPriority(samplingPriority);
 
-                if (traceId == null)
+                if (traceId.IsZero() &&
```

Comment:
Just to make sure I'm understanding this correctly, will both `default(TraceId)` and `TraceId.Zero` return true for IsZero()?

Reply:
Yes, thanks for pointing this out. I mean to come back to this and try to clean it up into a single construct everywhere. Do you have a preference?

```
if (traceId.IsZero())
if (isZero = traceId == TraceId.Zero)
if (isZero = traceId == default)
```

Reply:
I think `if (traceId == TraceId.Zero)` is the clearest, but we can see if there are other opinions on the team 😄

Reply:
Another vote for `if (traceId == TraceId.Zero)`

Reply:
I changed all `traceId.IsZero()` for `traceId == TraceId.Zero`.